From e676d2e38de0001bf033cda801dfdb62f54286e8 Mon Sep 17 00:00:00 2001
From: Justus Klausecker <justus@klausecker.de>
Date: Wed, 22 Oct 2025 01:03:43 +0200
Subject: [PATCH 017/133] std.mem.Allocator: fix resize doc comment
X-Developer-Signature: v=1; a=openpgp-sha256; l=926; i=xtex@astrafall.org;
 h=from:subject; bh=KRwVyLSQg2J2q5TOaxtkWvqTx6QdO+GjoG8hckUI+YY=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn7cn890Fh/sf5u/O3irxKmLzb2/zb4Efv0utWGBc/Ci
 lbu3dQU2jGRhUGMi8FSTJGlyLDBm1UnnV90WbkszBxWJpAh0iINDAwMDCwMfLmJeaVGOkZ6ptqG
 eoaGOkY6RgxcnAIw1bM2MjJMvnFceOLlYoO3j6a36Yv9WHcsw0Ks1Haf8PZpP4o8T/gHMfyzYst
 bIvXhoYXDcf3sx+ohB04KnnBqPmP6Zd28+YoMcUYMAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit a027fa8b8c03f8c325c4347500d0487c6d366f81)
---
 lib/std/mem/Allocator.zig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/std/mem/Allocator.zig b/lib/std/mem/Allocator.zig
index 897f1d2b31fe..35816a2fc4ff 100644
--- a/lib/std/mem/Allocator.zig
+++ b/lib/std/mem/Allocator.zig
@@ -290,7 +290,8 @@ fn allocBytesWithAlignment(
 /// It is guaranteed to not move the pointer, however the allocator
 /// implementation may refuse the resize request by returning `false`.
 ///
-/// `allocation` may be an empty slice, in which case a new allocation is made.
+/// `allocation` may be an empty slice, in which case `false` is returned,
+/// unless `new_len` is also 0, in which case `true` is returned.
 ///
 /// `new_len` may be zero, in which case the allocation is freed.
 pub fn resize(self: Allocator, allocation: anytype, new_len: usize) bool {
-- 
2.52.0

